engines/registry/lib/registry/engine.rb

```diff
@@ -1,6 +1,32 @@
+require 'fileutils'
+
 module Registry
   class Engine < ::Rails::Engine
     isolate_namespace Registry
     config.generators.api_only = true
+
+    config.after_initialize do
+      Api::Connect::V3::Systems::ActivationsController.class_eval do
+        before_action :handle_cache, only: %w[index]
+
+        def handle_cache
```

Comment:
```suggestion
        def update_registry_auth_cache
```
would be much more descriptive here